src/Files.App/Helpers/FtpHelpers.cs

```diff
@@ -67,13 +67,9 @@ public static ushort GetFtpPort(string path)
 		public static string GetFtpAuthority(string path)
 		{
 			path = path.Replace("\\", "/", StringComparison.Ordinal);
-			var schemaIndex = path.IndexOf("://", StringComparison.Ordinal) + 3;
-			var hostIndex = path.IndexOf("/", schemaIndex, StringComparison.Ordinal);
-
-			if (hostIndex == -1)
-				hostIndex = path.Length;
-
-			return path.Substring(schemaIndex, hostIndex - schemaIndex);
+			if (Uri.TryCreate(path, UriKind.Absolute, out var uri))
```

Comment:
Ftp path can contain username and pw: ftp://user@192.168.1.15:21/. Use Uri class to parse the Host/Authority as it supports user info unlike previous code.